peer.go

```diff
@@ -742,7 +742,36 @@ out:
 			}
 
 		case *lnwire.Error:
-			p.server.fundingMgr.processFundingError(msg, p.addr)
+			// Check if the error specifies an all-zero channel ID.
+			isZeroID := true
+			for _, b := range msg.ChanID {
+				if b != 0 {
+					isZeroID = false
+					break
+				}
+			}
+
+			switch {
+
+			// In the case of an all-zero channel ID we want to
+			// forward the error to all channels with this peer.
+			case isZeroID:
```

Comment:
nit: can we switch on `msg.ChanID == lnwire.ChannelID` and remove above check?

Reply:
We'll need to be careful here. An all zero error is meant to indicate that the error applies to the entire connection, not necessarily to all channels. In the near future, the `fail` message in the link will actually _force close_ the channel. 

This section in the spec:
```
when channel_id is 0:
MUST fail all channels.
MUST close the connection.
```

is a bit _too_ strong IMO. We should be cautious and only fail out channels if _absolutely_ necessary.

Reply:
Well, if the spec says we **must** fail the channel, then we **really should**, right? :P

If the sending node is sending an all-zero error message, than it is also required to fail out all channels, so don't guaranteed we could recover from it anyways.

Reply:
Sure, but from a game-ability perspective...I can just send an node a particular error to _force_ it to close all channels incurring the worst case delay. Worse, I can now orchestrate a synchronized channel closure across many nodes, just by sending this particular error.

Reply:
I agree. However, a malicious node could always just send individual messages for each channel, causing the same behaviour. 

Currently we are not actually closing the channels on-chain. I say when we add that we will be careful not to force close if we receive `lnwire.Error` from the peer.